### PR TITLE
Add 'one required flag' group

### PR DESCRIPTION
**Pull request by marevers**
This PR addresses #1761 and #1891 - it adds a new flag group which can be used to group a set of flags of which at least one is required. If the related command is invoked and none of the flags in the group are set, the user is informed of the flag group (e.g. with two flags called 'path' and 'id'):

```
Error: at least one of the flags in the group [path id] is required
```

It is used like the other flag groups (MarkFlagsMutuallyExclusive / MarkFlagsRequiredTogether):

```
getCmd.AddCommand(getEntryCmd)
getEntryCmd.MarkFlagsOneRequired("path", "id")
```

It can also be combined with MarkFlagsMutuallyExclusive with the same set of flags to enforce only one flag (and not more than one).

### CODE REVIEW

**Review by marckhouzam — COMMENTED** [review]
Thanks @marevers, this looks great.
I've added this to the planned features for the next release.

1. Add you mentioned, could you add some tests please?
2. Could you also update `enforceFlagGroupsForCompletion()`. I think what we need there is that: if none of the flags in the new required group are present, all of them should be marked as required so the completion suggests all of them, but as soon as one of them is present, we do nothing more.

**Comment by marckhouzam** [comment]
Nicely done @marevers !  Thanks.
I want to run a bit more tests, but this should be ready to merge soon.

**Review by marckhouzam — COMMENTED** [review]
Everything is fine.
One last thing to ensure quality, can you also add a test like `TestCompletionForGroupedFlags` and `TestCompletionForMutuallyExclusiveFlags`
Then we can merge.

Thanks

**Comment by marckhouzam** [comment]
@marevers Sorry for the confusion.  The `TestCompletionForGroupedFlags` and `TestCompletionForMutuallyExclusiveFlags` already exist in the `completions_test.go` file.  You can do something very similar for a new `TestCompletionForOneRequiredGroupFlags`

**Comment by marevers** [comment]
@marckhouzam Thanks for the clarification. I added a new test function `TestCompletionForOneRequiredGroupFlags` based on `TestCompletionForGroupedFlags` with very similar subtests but also verifying that when any flag in the one-required group is present, other flags will not be suggested.

**Review by marckhouzam — APPROVED** [review]
LGTM
Thanks @marevers !

**Comment by marckhouzam** [comment]
@marevers I was about to press merge, and then I remembered we need to update the doc.  It should be very quick for you.  Can you update this section please: https://github.com/spf13/cobra/blob/main/site/content/user_guide.md#flag-groups

**Comment by marevers** [comment]
@marckhouzam Okay, that should do it. I had to resync my fork because `user_guide.md` had been moved since I made that fork.

File: site/content/user_guide.md [hunk 1]
```diff
@@ -349,7 +349,16 @@ rootCmd.Flags().BoolVar(&ofYaml, "yaml", false, "Output in YAML")
 rootCmd.MarkFlagsMutuallyExclusive("json", "yaml")
 ```
 
-In both of these cases:
+If you want to require least one flag from a group to be present, you can use `MarkFlagsOneRequired`.
```
Comment (marckhouzam) [inline site/content/user_guide.md:352]:
```suggestion
If you want to require at least one flag from a group to be present, you can use `MarkFlagsOneRequired`.
```

**Review by marckhouzam — CHANGES_REQUESTED** [review]
One typo and we're good to go.

**Comment by marevers** [comment]
> One typo and we're good to go.

Fixed!

**Review by marckhouzam — APPROVED** [review]
LGTM
Thanks @marevers !
There’s already scenarios where this new feature can be used in the CLI I work on. This will be great!

**Pull request merged**